xarray/core/datatree.py

```diff
@@ -77,7 +73,7 @@
 # """
 
 
-T_Path = Union[str, NodePath]
+T_Path = str | NodePath
```

Comment:
Looks like `mypy` and the Python 3.9 tests don't like this change. I can revert it if that sounds best?

Reply:
Yes that syntax won't work on 3.9 so we can't use it for now.

Reply:
I think you can add the pragma at the top? ```from __future__ import annotations```. Maybe it's technically a pragma.

Reply:
I think [it was already there](https://github.com/pydata/xarray/blob/main/xarray/datatree_/datatree/datatree.py#L1), so I'm guessing the custom type here is not covered by `__future__.annotations` in the same way that type hints are?

Reply:
that's what I get for guessing without looking.